kernel/src/log_segment.rs

```diff
@@ -206,6 +207,61 @@ impl LogSegment {
         LogSegment::try_new(listed_files, log_root, end_version)
     }
 
+    #[allow(unused)]
+    /// Constructs a [`LogSegment`] to be used for timestamp conversion. This [`LogSegment`] will
+    /// consist only of contiguous commit files up to `end_version` (inclusive). If present,
+    /// `limit` specifies the maximum length of the returned log segment. The log segment may be
+    /// shorter than `limit` if there are missing commits.
+    ///
+    // This lists all files starting from `end-limit` if `limit` is defined. For large tables,
+    // listing with a `limit` can be a significant speedup over listing _all_ the files in the log.
+    pub(crate) fn for_timestamp_conversion(
+        storage: &dyn StorageHandler,
+        log_root: Url,
+        end_version: Version,
+        limit: Option<NonZero<usize>>,
+    ) -> DeltaResult<Self> {
+        // Compute the version to start listing from.
+        let start_from = limit
+            .map(|limit| match NonZero::<Version>::try_from(limit) {
+                Ok(limit) => Ok(Version::saturating_sub(end_version, limit.get() - 1)),
+                _ => Err(Error::generic(format!(
+                    "Invalid limit {limit} when building log segment in timestamp conversion",
+                ))),
+            })
+            .transpose()?;
+
+        let mut contiguous_commits: Vec<ParsedLogPath> =
+            limit.map_or_else(Vec::new, |limit| Vec::with_capacity(limit.get()));
+
+        for commit_res in list_log_files(storage, &log_root, start_from, Some(end_version))? {
+            let commit = match commit_res {
+                Ok(file) if file.is_commit() => file,
```

Comment:
I'd like to get rid of the `is_commit` method, so we don't have multiple ways of doing the same thing. So maybe:
```suggestion
                Ok(file) if matches!(file.file_type, LogPathFileType::Commit) => file,
```